### ISSUE TITLE
[Feat]: Web-based GPG pin entry for signing git commits

### ISSUE THREAD

**Issue by MitchTalmadge** [coder/code-server#5271]
## What is your suggestion?
When signing git commits with GPG, where the GPG key exists on the server, display a prompt that allows for entry of the GPG pin.

## Why do you want this feature?
If you currently try to commit from the UI when using GPG signing, you get an error: 
![image](https://user-images.githubusercontent.com/8848080/173770079-9483009b-b003-4f39-a2bf-e3bb05da685f.png)

```
error: gpg failed to sign the data
fatal: failed to write commit object
```

...but if you use the terminal, it works fine:
![image](https://user-images.githubusercontent.com/8848080/173770228-b0f3fc5f-3463-4d6e-b5ec-6f4789068a9f.png)

But this requires me to open the terminal every time I want to commit. I would prefer to use the nice UI.

## Are there any workarounds to get this functionality today?
Not that I know of. I could just be doing something entirely wrong.

## Are you interested in submitting a PR for this?
I wouldn't really know where to begin here, unfortunately, and lack the time at the moment to figure it out.

**Comment by jsjoeio** [comment]
Is this a bug? Does this happen in VS Code/Codespaces?

**Comment by Sominemo** [comment]
> Is this a bug? Does this happen in VS Code/Codespaces?

Native VS Code triggers pinentry program from system. In case of code server, it would be cool if it would have a web-based fallback for it.

**Comment by code-asher** [comment]
Agreed that it would be cool to handle this in the UI and it should be pretty straightforward, we just need to implement a pinentry script that talks to code-server via the IPC socket.  Probably involves adding some new message types but I think it would be relatively easy.

Likely this reproduces in Codespaces though so I would lean toward raising this upstream.

No workarounds other than the ones mentioned come to mind unfortunately.

**Comment by Sominemo** [comment]
Just to be sure, tried to reproduce it in Codespaces, and it has exactly the same behavior.

**Comment by jsjoeio** [comment]
@Sominemo that's great! do you mind opening an issue upstream?

**Comment by Sominemo** [comment]
Seems like upstream has a related issue already: https://github.com/microsoft/vscode-remote-release/issues/1022
There's also a nice workaround that suggests using [this extension][1], mentioned in comment https://github.com/microsoft/vscode-remote-release/issues/1022#issuecomment-1123209855

[1]: https://github.com/wdhongtw/vscode-gpg-indicator
